### DISCUSSION TITLE
Support for onewire (1-wire) in Berry Scripting language

### DISCUSSION THREAD

**Discussion by TomsTek** [arendst/Tasmota#12857] [Ideas]
Hello Community,
the Berry Scripting language is a very cool tool.

Is ist possible to support onewire interface with Berry?
Not all special, own or new Sensors can supported with standard drivers.

An Idea is, to create an connection to the OneWire.h Arduino Library.

Thank you for Feedback.

**Comment by s-hadinger** [comment]
Hi. This shouldn't be too difficult. There is already a mapping for I2C, I suppose that OneWire is similar. 

My only problem is that I don't have any OneWire sensor so I can't test.

**Reply by TomsTek** [reply]
Hi,
I could test the feature, but i have no overview over required code changes.

I know a bit the structure of the tasmota project, but not the komplex connectivity in the whole project.

**Reply by s-hadinger** [reply]
No problem. I will do a mapping in the coming weeks. What device do you have?

**Reply by TomsTek** [reply]
Thats Great.
At the first step i would test an DS18b20 Temperature Sensor. So i know the sensor works and I know how it acts.
Later I would use an emulated DS2431 or DS2408 to connect own Devices. Further every onewire device is usable.

Offtopic: Last weekend i build an I²C-Driver for an TMP117 temperature sensor an I was fascinated how easy it runs.

If I can support you, let me know how I can do. I don't know where you live, by any chance I can send you an Sensor?

**Reply by s-hadinger** [reply]
Interesting. Will you share your TMP117 driver?

**Reply by TomsTek** [reply]
I would like to make the driver a little nicer and add checks. Then I can make it available too.

**Reply by s-hadinger** [reply]
OneWire support is there but I can't test it. Could you try?
#12955

Compile with `#define USE_ONEWIRE`

**Reply by TomsTek** [reply]
Nice. I will see to make first test the next days.

**Reply by TomsTek** [reply]
I have forgotten to share may TMP117 (I²C) Berry driver.
https://github.com/TomsTek/tasmota-berry-TMP117-driver
